### PR TITLE
[auto parallel] unify dynamic and static recompute api

### PR DESCRIPTION
**Pull request by FeixLiu**
### PR types
Others

### PR changes
Others

### Description

PCard-73145
unify dynamic and static recompute api

### CODE REVIEW

**Review by LiYuRio — DISMISSED** [review]
LGTM

File: python/paddle/distributed/auto_parallel/api.py [hunk 1]
```diff
@@ -525,6 +538,26 @@ def pipeline(self):
...
         return self._pipeline
 
+    @property
+    def recompute(self):
+        """
+
+        ``recompute`` is used to configure the recompute strategy in training,
+        containing following configs:
+
+            ``enable`` (bool): whether to enable recompute. Default: False.
```
Comment (sunzhongkai588) [inline python/paddle/distributed/auto_parallel/api.py]:
```suggestion
           -  ``enable`` (bool): whether to enable recompute. Default: False.
```

**Review by sunzhongkai588 — COMMENTED** [review]
好像 code style 也没过，用 pre commit 自动修一下吧
图片

**Review by sunzhongkai588 — DISMISSED** [review]
LGTM for docs

**Review by LiYuRio — DISMISSED** [review]

File: python/paddle/distributed/auto_parallel/api.py [hunk 2]
```diff
@@ -525,6 +538,26 @@ def pipeline(self):
...
+        Examples:
+            .. code-block:: python
+
+                >>> import paddle
+                >>> import paddle.distributed as dist
+                >>> strategy = dist.Strategy()
+                >>> strategy.recompute.enable = True
+
+        """
+        return self._recompute
```
Comment (zhiqiu) [inline python/paddle/distributed/auto_parallel/api.py]: As discussed, remove this config in next pr

**Review by zhiqiu — DISMISSED** [review]
LGTM

**Review by LiYuRio — APPROVED** [review]

**Pull request merged**